app/jobs/regular/emit_web_hook_event.rb

```diff
@@ -2,23 +2,30 @@
 
 module Jobs
   class EmitWebHookEvent < Jobs::Base
+    # Required option:
+    #   model                   - The symbol represents the model class
+    #
+    # Acceptable options:
+    #   load_record             - A lambada for loading the records
+    #   serializer_name         - A serializer class name
+    def self.web_hook_serialization_properties
+      @@web_hook_serialization_properties ||= [
+        { model: :topic, load_record: ->(topic_id) { TopicView.new(topic_id, Discourse.system_user) }, serializer_name: "WebHookTopicViewSerializer" },
+        { model: :post, serializer_name: "WebHookPostSerializer" },
+        { model: :user, serializer_name: "WebHookUserSerializer" }
+      ]
+    end
+
     def execute(args)
       raise Discourse::InvalidParameters.new(:web_hook_id) unless args[:web_hook_id].present?
       raise Discourse::InvalidParameters.new(:event_type) unless args[:event_type].present?
 
       args = args.dup
 
-      if args[:topic_id]
-        args[:topic_view] = TopicView.new(args[:topic_id], Discourse.system_user)
-      end
-
-      if args[:post_id]
-        # deleted post so skip
-        return unless args[:post] = Post.find_by(id: args[:post_id])
-      end
-
-      if args[:user_id]
-        return unless args[:user] = User.find_by(id: args[:user_id])
+      Jobs::EmitWebHookEvent.web_hook_serialization_properties.each do |property|
+        type_id = "#{property[:model]}_id".to_sym
+        next unless args[type_id]
+        return unless (args[property[:model]] = (property[:load_record] || ->(id) { property[:model].to_s.classify.constantize.find_by(id: id.to_i) }).call(args[type_id]))
```

Comment:
I'm not sure if I like the use of a `lambda` here. It makes the code really hard to read.